packages/eslint-plugin-kolibri/tests/lib/rules/no-undefined-translator-keys.spec.js

```diff
@@ -0,0 +1,723 @@
+'use strict';
+
+const path = require('path');
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../../../lib/rules/no-undefined-translator-keys');
+
+// Compute test file path dynamically so it works on any machine
+const testFilePath = path.join(__dirname, 'test.js');
+
+const tester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2020,
+    sourceType: 'module',
+  },
+});
+
+tester.run('no-undefined-translator-keys', rule, {
+  valid: [
+    // ========================================
+    // Valid Cases
+    // ========================================
+
+    // 1. Basic valid destructuring
+    {
+      code: `
+        const translator = createTranslator('NS', { validKey: 'msg' });
+        const { validKey$ } = translator;
+      `,
+    },
+
+    // 2. Multiple valid keys
+    {
+      code: `
+        const translator = createTranslator('NS', { key1: 'a', key2: 'b', key3: 'c' });
+        const { key1$, key2$, key3$ } = translator;
+      `,
+    },
+
+    // 3. Object format messages
+    {
+      code: `
+        const translator = createTranslator('NS', {
+          key1: { message: 'msg', context: 'ctx' },
+          key2: { message: 'msg2' }
+        });
+        const { key1$, key2$ } = translator;
+      `,
+    },
+
+    // 4. Exported translator with destructuring in same file
+    {
+      code: `
+        export const translator = createTranslator('NS', { validKey: 'msg', anotherKey: 'msg2' });
+        const { validKey$ } = translator;
+      `,
+    },
+
+    // 5. No destructuring (direct usage)
+    {
+      code: `
+        const translator = createTranslator('NS', { key: 'msg' });
+        translator.$tr('key');
+      `,
+    },
```

Comment:
It doesn't seem like this rule is handling these `$tr()` cases. I think adding this case in this test file may be a bit misleading.

Reply:
Yeah - I think this is meant to be more of a 'doesn't flag a case it shouldn't' - but I am not sure it's entirely necessary.